pkg/reconciler/knativeeventing/common/config_maps.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	eventingv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
+)
+
+// ConfigMapTransform updates the ConfigMap with the values specified in operator CR
+func ConfigMapTransform(instance *eventingv1alpha1.KnativeEventing, log *zap.SugaredLogger) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		// Let any config in instance override everything else
+		if u.GetKind() == "ConfigMap" {
+			if data, ok := instance.Spec.Config[u.GetName()[len(`config-`):]]; ok {
```

Comment:
could stripping off the 'config-' ever lead to collisions?

Reply:
As long as the CM name starts with "config-", it should be fine.
Luckily, all CMs in knative start with "config-".

Reply:
Alright :) Do we have that constraint documented anywhere?

I'd just like it to be formalized if we're depending on it here.